search-enterprise/latest/config/transactional.md

```diff
@@ -0,0 +1,382 @@
+
+TMDQ supports:
+
+* `query`
+* `paging`
+* `include`
+* `includeRequest`
+* `fields`
+* `sort`
+* `defaults`
+* `filterQueries`
+* `scope` (single)
+* `limits` for permission evaluation
+
+The default limits for permission evaluation will restrict the results returned from TMDQ based on both the number of results processed and time taken. These can be increased, if required.
+
+The Public API does not support TMDQ for:
+
+* `templates`
+* `localisation` and `timezone`
+* `facetQueries`
+* `facetFields`
+* `facetIntervals`
+* `pivots`
+* `stats`
+* `spellcheck`
+* `highlight`
+* `ranges facets`
+
+Some of these will be ignored and produce transactional results; others will fail and be eventual.
+
+The Public API ignores the SQL select part of a CMIS query and generate the results as it would do for AFTS.
+
+### CMIS QL & TMDQ
+
+For CMIS QL, all expressions except for `CONTAINS()`, `SCORE()`, and `IN_TREE()` can now be executed against the database. Most data types are supported except for the CMIS uri and html types. Strings are supported but only if there are 1024 characters or less in length.
+
+In Alfresco One 5.0, `OR`, decimal, and boolean types were not supported; it is only from Alfresco One 5.1 onwards that they are supported.
+
+Primary and secondary types are supported and require inner joins to link them together. You can skip joins to secondary types from the fetch in CMIS using the public API. You would need an explicit `SELECT` list and supporting joins from a CMIS client. You still need joins to secondary types for predicates and ordering. As CMIS SQL supports ordering as part of the query language, you have to do it there and not via the Public API sort.
+
+For multi-valued properties, CMIS QL supports `ANY` semantics from SQL 92. A query against a multi-lingual property, such as title or description, is treated as multi-valued and may match in any language. In the results, you will see the best value for your locale, which may not match the query. Also, ordering will consider any value.
+
+### UPPER() and LOWER()
+
+`UPPER()` and `LOWER()` functions were in early drafts for the CMIS 1.0 specification, but were subsequently dropped. These are not part of the CMIS 1.0 or 1.1 specifications. They are not supported in TMDQ.
+
+### Alfresco FTS QL & TMDQ
+
+It is more difficult to write AFTS queries that use TMDQ as the default behaviour is to use full text queries for text. These cannot go against the database. Also, special fields like `SITE` and `TAG` that are derived from the structure will not go to the database. `TYPE`, `ASPECT` and the related exact matches work fine with TMDQ. All property data types are fine but strings should be less than 1024 characters in length. Text queries have to be prefixed with `=` to avoid full text search. Additionally, `PARENT` is supported but `OR` is supported from Alfresco One 5.1 onwards.
```

Comment:
```suggestion
It is more difficult to write AFTS queries that use TMDQ as the default behaviour is to use full text queries for text and full text queries cannot be served by the database. Also, special fields like `SITE` and `TAG` that are derived from paths or other nodes will not be handled by the database. `TYPE`, `ASPECT` and the related exact matches will work with TMDQ. All property data types are fine but strings should be less than 1024 characters in length. Text queries have to be prefixed with `=` to avoid full text search.
```